client/python/lib/ovmsclient/tfs_compat/grpc/responses.py

```diff
@@ -15,26 +15,66 @@
 #
 
 from tensorflow_serving.apis.get_model_status_pb2 import ModelVersionStatus
+from tensorflow_serving.apis import get_model_metadata_pb2
+from tensorflow.core.framework.types_pb2 import DataType
 
 from ovmsclient.tfs_compat.base.responses import PredictResponse, ModelMetadataResponse, ModelStatusResponse
 
 class GrpcPredictResponse(PredictResponse):
     pass
 
 class GrpcModelMetadataResponse(ModelMetadataResponse):
-    
+
     def to_dict(self):
-        raise NotImplementedError
+        result_dict = {}
+
+        signature_def = self.raw_response.metadata['signature_def']
+        signature_map = get_model_metadata_pb2.SignatureDefMap()
+        signature_map.ParseFromString(signature_def.value)
+        serving_default = signature_map.ListFields()[0][1]['serving_default']
+        
+        serving_inputs = serving_default.inputs
+        input_blobs = {key : {} for key in serving_inputs.keys()}
+        tensor_shape = {key: serving_inputs[key].tensor_shape for key in serving_inputs.keys()}
+
+        input_blobs_dict = {}
+        for input_blob in input_blobs:
+            inputs_shape = [d.size for d in tensor_shape[input_blob].dim]
+            tensor_dtype = serving_inputs[input_blob].dtype
+            input_blobs_dict[input_blob] = dict([
+                ("shape", inputs_shape),
+                ("dtype", DataType.Name(tensor_dtype))
+            ])
```

Comment:
I think we can simplify this part as well as the similar one below for the output.
What if we wouldn't create `input_blobs` and `tensor_shape` dicts?
We could just have:
```
       inputs_metadata = {}
        for input_name, input_info in model_signature.inputs:
            input_shape = [d.size for d in input_info.tensor_shape.dim]
            inputs_metadata[input_name] = dict([
                ("shape", input_shape),
                ("dtype", DataType.Name(input_info.dtype))
            ])
```

This code snippet assumes, you switch from `serving_default` to `model_signature` name in assignment in line 34:
`serving_default = signature_map.ListFields()[0][1]['serving_default']`

That would look better imho.

Reply:
This change causes 3 tests to fail with:
```ValueError: not enough values to unpack (expected 2, got 1)```

Reply:
with ```for input_name, input_info in model_signature.inputs.items():``` now works as intended